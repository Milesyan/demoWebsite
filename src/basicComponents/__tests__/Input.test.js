/* eslint-disable no-undef */ /*eslint-disable react/react-in-jsx-scope*/

import { Input } from '../index';

const setup = overrideProps => {
  const initProps = {
    onEmitUpdates: jest.fn(),
    errMsg: null,
    placeholder: "Place Holder",
  };
  const props = { ...initProps, ...overrideProps};
  const wrapper = shallow(
    <Input {...props}/>
  );
  return { wrapper, props };
}

describe('Input snapshots', () => {

  it('should match snapshot with placeholder', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match snapshot with error message', () => {
    const { wrapper } = setup({errMsg: 'Example Error'});
    expect(wrapper).toMatchSnapshot();
  });

});

describe('Input functions', () => {

  it('should not emit text updates for initial state', () => {
    const { wrapper, props } = setup();
    wrapper.find('input').simulate('change', "EXAMPLE CHANGE 1");
    expect(props.onEmitUpdates).not.toBeCalled();
  });  

  it('should not emit text updates if set shouldEmitUpdates to be true', () => {
    const { wrapper, props } = setup({shouldEmitUpdates: true});
    wrapper.find('input').simulate('change', "EXAMPLE CHANGE 2");
    expect(props.onEmitUpdates).toHaveBeenLastCalledWith("EXAMPLE CHANGE 2");
  });  

  it('should not emit text updates if the input component onBlur has been called', () => {
    const { wrapper, props } = setup({shouldEmitUpdates: false});
    wrapper.find('input').simulate('blur', "EXAMPLE BLUR 3");
    expect(props.onEmitUpdates).toHaveBeenLastCalledWith("EXAMPLE BLUR 3");
    wrapper.find('input').simulate('change', "EXAMPLE CHANGE 4");
    expect(props.onEmitUpdates).toHaveBeenLastCalledWith("EXAMPLE CHANGE 4");
  });  
  
});