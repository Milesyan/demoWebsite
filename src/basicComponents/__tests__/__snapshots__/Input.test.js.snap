// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshots should match snapshot with error message 1`] = `
<div
  className="container"
>
  <input
    className="dialogInput errorInput"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Place Holder"
  />
  <div
    className="errorMsg"
  >
    Example Error
  </div>
</div>
`;

exports[`Input snapshots should match snapshot with placeholder 1`] = `
<div
  className="container"
>
  <input
    className="dialogInput"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Place Holder"
  />
</div>
`;
